pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -28,8 +30,17 @@
 from pytorch_lightning.callbacks.timer import Timer
 from pytorch_lightning.utilities.enums import ModelSummaryMode
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _PYTHON_GREATER_EQUAL_3_8_0
 from pytorch_lightning.utilities.rank_zero import rank_zero_deprecation, rank_zero_info
 
+if _PYTHON_GREATER_EQUAL_3_8_0:
+    import importlib.metadata
+else:
+    import pkg_resources
```

Comment:
Should we import locally at the branch you have in `_configure_external_callbacks` instead?

Then, we only branch in one place and adds one less import at startup: #12786

Reply:
Unfortunately, if we do that, I can't anymore patch the functions for the test.

Reply:
You should be able to, by using `mock.patch` instead of `monkeypatch.setattr` since you cannot reference the local variable

Reply:
[f2ea314](https://github.com/PyTorchLightning/pytorch-lightning/pull/12739/commits/f2ea314be0bb78950f97139e60442fd0f83b9f67)

Reply:
thanks for the help here!